### PR TITLE
Remote monitoring

### PR DESCRIPTION
**Pull request by brushknight**

### CODE REVIEW

File: jmx-config.yml.k8s_sample [hunk 1]
```diff
@@ -20,6 +20,7 @@
           JMX_USER: admin
           JMX_PASS: admin
           COLLECTION_FILES: ${config.path}
+          REMOTE_MONITORING: true
```
Comment (cristianciutea) [inline jmx-config.yml.k8s_sample:23]: can you please add the same description as we have on other integrations?

File: src/request.go [hunk 2]
```diff
@@ -102,7 +102,19 @@ func insertDomainMetrics(eventType string, domain string, beanAttrVals []*beanAt
 	// Create an entity for the domain
 	var e *integration.Entity
 	var err error
-	if args.LocalEntity {
+	if args.RemoteMonitoring {
+		if args.ConnectionURL != "" {
```
Comment (cristianciutea) [inline src/request.go:106]: can we extract this into a function that returns the remoteEntity?

File: src/request.go [hunk 3]
```diff
@@ -102,7 +102,19 @@ func insertDomainMetrics(eventType string, domain string, beanAttrVals []*beanAt
...
+			e, err = i.Entity(fmt.Sprintf("%s:%s", domain, args.ConnectionURL), "jmx-domain")
```
Comment (cristianciutea) [inline src/request.go:107]: We could use the specifications from here to parse the ConnectionUrl https://docs.oracle.com/javase/7/docs/api/javax/management/remote/JMXServiceURL.html

File: jmx-config.yml.k8s_sample (hunk 1 above)
Comment (brushknight) [reply jmx-config.yml.k8s_sample:23]: sure

File: src/request.go (hunk 2 above)
Comment (brushknight) [reply src/request.go:106]: will do

File: src/request.go (hunk 3 above)
Comment (brushknight) [reply src/request.go:107]: port can be omitted and uri part can be the only difference, so what it we use this part `[host[:port]][url-path]` as a replacement for host:port in this case?

File: src/request.go [hunk 4]
```diff
@@ -345,3 +358,20 @@ func inferMetricType(s interface{}) metric.SourceType {
...
+	return i.Entity(fmt.Sprintf("%s:%s", domain, suffix), "jmx-domain")
+}
+
+// getConnectionUrlSAP extracts last part that describes connection string,
+// in case that it can't extract SAP part, will return full connection URL
+// ref: https://docs.oracle.com/javase/7/docs/api/javax/management/remote/JMXServiceURL.html
+func getConnectionUrlSAP(connectionUrl string) string {
+	r := strings.Split(connectionUrl, "://")
+
+	if len(r) == 0 {
```
Comment (brushknight) [inline src/request.go:372]: probably this should be done if not 3 elements in the slice

File: src/request.go (hunk 3 above)
Comment (cristianciutea) [reply src/request.go]: we should make sure, there is no issue if there are special url characters into the entityName. An alternative would be to get host:port by checking the ":" location and then remove everything after the first "/"

**Comment by cristianciutea** [comment]
can you please check the failure on static analysis check?

File: src/request.go [hunk 5]
```diff
@@ -102,7 +102,19 @@ func insertDomainMetrics(eventType string, domain string, beanAttrVals []*beanAt
...
+			if err != nil {
+				return err
+			}
+		} else {
+			e, err = newRemoteEntity(domain, fmt.Sprintf("%s:%s", host, port), i)
+			if err != nil {
+				return err
+			}
+		}
+	} else if args.LocalEntity {
```
Comment (rubenruizdegauna) [inline src/request.go:117]: not a big issue but I would simplify this to:

```go
	if args.RemoteMonitoring {
		url := fmt.Sprintf("%s:%s", host, port)
		if args.ConnectionURL != "" {
			url = getConnectionUrlSAP(args.ConnectionURL)
		}
		e, err = newRemoteEntity(domain, url, i)
		if err != nil {
			return err
		}
	}
```

**Review by rubenruizdegauna — APPROVED** [review]
LGTM! just small suggestion

File: src/request.go (hunk 5 above)
Comment (brushknight) [reply src/request.go]: thank you, this makes code simpler.

**Review by rubenruizdegauna — APPROVED** [review]
LGTM!

**Pull request merged**